### PR TITLE
Use a ticker instead of a while loop.

### PR DESCRIPTION
**Pull request by TheSeeker**
Probably doing it wrong, but it seems to work.

### CODE REVIEW

**Comment by xor-freenet** [comment]
> \<xor-freenet\> ArneBab_: you really shouldn't use ticker directly but have a look at the BackgroundJob stuff of WoT... it basically is a wrapper around Ticker to make it behave in a sane fashion.
> \<xor-freenet\> here's the main reason: https://freenet.mantishub.io/view.php?id=6423
> \<xor-freenet\> ArneBab: overall the approach of "query the database every X minutes for whether we need to do something" is one of the main reasons why WoT grew so slow, and i would really recommend to have an event-driven approach instead, i.e. "whatever queues the insert should trigger execution of the insert loop"...
> \<xor-freenet\> ArneBab:  which would fit together very well with the BackgroundJob stuff as it prevents multiple executions of the thread if there are multiple executions of it requested before the delay expired.

File: src/plugins/Sharesite/Inserter.java [hunk 1]
```diff
@@ -25,65 +27,41 @@
...
-					// wait until at most 5 minutes before end of the hour
-					Integer waitTime = (int)(Math.random() * (55 - currentMinute));
-					try {
-						if (waitTime > 0) {
-							Thread.sleep((waitTime * 60 * 1000));
-						}
-					} catch (InterruptedException e) {
-						// break the loop when interrupted, we are only waiting
-						break;
+		Integer currentHour = Calendar.getInstance(TimeZone.getTimeZone("UTC"), Locale.US)
```
Comment (xor-freenet) [inline src/plugins/Sharesite/Inserter.java:42]: This should instead use freenet.support.CurrentTimeUTC.get()

**Comment by ArneBab** [comment]
Nice, thank you! This should get me through my mental overload.

I’ll incorporate the feedback by xor and test it. As far as I understand it, this keeps ticking only while there are freesites to insert. The ticker is then restarted when something queues a new insert.

**Comment by ArneBab** [comment]
Merged with a fix to run initial inserts instantly. Thank you again!

**Pull request merged**